collective/weightedportlets/editmanager.py

```diff
@@ -35,10 +35,12 @@ def __call__(self):
             return self.weighted_message('Error saving data.')
 
         info = unhashPortletInfo(portlethash)
+        # key needs to be converted to str as unicode paths are not accepted
+        # in restrictedTraverse called in assignment_mapping_from_key
         assignments = assignment_mapping_from_key(self.context,
                                                   info['manager'],
                                                   info['category'],
-                                                  info['key'])
+                                                  info['key'].encode())
```

Comment:
I think this will fail if the key has non-ASCII characters. How about `.encode('utf8')`?

Reply:
Thanks for the comment (and for your very useful weightedportlets package). I agree it'd be more foolproof that way, will do it. There is a similar code in plone.app.portlets/exportimport/portlets.py in _initAssignmentNode(), not sure if utf-8 needs to be specified there as well.

Reply:
I see. I suppose it would be best to be explicit about the encoding there too, but I guess if it is there and hasn't been a problem, then maybe non-ASCII characters aren't used in the `key` in practice. I'm fine with not worrying about it until something actually breaks.